### PR TITLE
[IMP] SUITE account payment return: Improvements to suite

### PR DESCRIPTION
**Pull request by carlosdauden**
- Add raise "You can not be removed a payment return if state is 'Done'".
- New SEPA return code.
- Add check to match returns after import.
- Spanish translation.

@Tecnativa

### CODE REVIEW

File: account_payment_return/models/payment_return.py [hunk 1]
```diff
@@ -108,6 +108,13 @@ def _prepare_invoice_returned_cancel_vals(self):
         return {'returned_payment': False}
 
     @api.multi
+    def unlink(self):
+        if self.filtered(lambda x: x.state == 'done'):
+            raise UserError(_(
+                "You can not be removed a payment return if state is 'Done'"))
```
Comment (pedrobaeza) [inline account_payment_return/models/payment_return.py]: Remove `be`

File: account_payment_return_import/i18n/es.po [hunk 2]
```diff
@@ -83,12 +91,14 @@ msgid "Cancel"
...
 #. module: account_payment_return_import
-#: code:addons/account_payment_return_import/wizard/payment_return_import.py:141
+#: code:addons/account_payment_return_import/wizard/payment_return_import.py:144
 #, python-format
 msgid ""
 "Could not make sense of the given file.\n"
 "Did you install the module to support this type of file?"
-msgstr "No se pudo hacer con sentido de el archivo dado.\n¿Instalaste el modulo para soportar este tipo de archivo?"
+msgstr ""
+"No se pudo hacer con sentido de el archivo dado.\n"
```
Comment (pedrobaeza) [inline account_payment_return_import/i18n/es.po]: Trato de usted, no de tú

File: account_payment_return_import_sepa_pain/data/unpaid_reason_codes.xml [hunk 3]
```diff
@@ -42,6 +42,10 @@
         <field name="code">MD01</field>
         <field name="name">No valid Mandate</field>
     </record>
+    <record id="unpaid_reason_MD06" model="payment.return.reason">
+        <field name="code">MD06</field>
+        <field name="name">Disputed authorised transaction</field>
```
Comment (pedrobaeza) [inline account_payment_return_import_sepa_pain/data/unpaid_reason_codes.xml:47]: This isn't translated?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: account_payment_return_import/i18n/es.po (hunk 2 above)
Comment (carlosdauden) [reply account_payment_return_import/i18n/es.po]: Leo cosas contradictorias al respecto, por eso nunca lo tengo claro...

File: account_payment_return_import_sepa_pain/data/unpaid_reason_codes.xml (hunk 3 above)
Comment (carlosdauden) [reply account_payment_return_import_sepa_pain/data/unpaid_reason_codes.xml:47]: https://github.com/OCA/account-payment/pull/107/files#diff-e37f5160827a396976dac1201391ad64R61

Comment (pedrobaeza) [reply account_payment_return_import_sepa_pain/data/unpaid_reason_codes.xml:47]: OK

**Comment by pedrobaeza** [comment]
Include a test for unlinking returns for not dropping coverage

**Comment by carlosdauden** [comment]
@pedrobaeza changes done

**Comment by carlosdauden** [comment]
cc @sergio-teruel

**Comment by pedrobaeza** [comment]
Please don't squash while the review process. You only need to squash before merging (when the PR is approved), or it's more difficult to track changes from one review to the next. Reviewer can indicate you if you need to squash or not.

**Comment by pedrobaeza** [comment]
You have to test also a case where there's no error and the payment can be removed.

**Review by pedrobaeza — APPROVED** [review]
Thanks for the changes.

To the merger: squash on merging

**Review by rafaelbn — APPROVED** [review]
OK! Thanks

**Review by sergio-teruel — APPROVED** [review]
Ok..

File: account_payment_return/models/payment_return.py [hunk 4]
```diff
@@ -108,6 +108,13 @@ def _prepare_invoice_returned_cancel_vals(self):
         return {'returned_payment': False}
 
     @api.multi
+    def unlink(self):
+        if self.filtered(lambda x: x.state == 'done'):
+            raise UserError(_(
+                "You can not remove a payment return if state is 'Done'"))
+        return super(PaymentReturn, self).unlink()
```
Comment (yajo) [inline account_payment_return/models/payment_return.py:115]: I'd implement this as a [global ir.rule](https://www.odoo.com/documentation/8.0/reference/security.html#record-rules) instead.

**Review by yajo — CHANGES_REQUESTED** [review]

File: account_payment_return/models/payment_return.py (hunk 4 above)
Comment (carlosdauden) [reply account_payment_return/models/payment_return.py:115]: It is an option, but usually this is the way used and possibly better performance.

Comment (pedrobaeza) [reply account_payment_return/models/payment_return.py:115]: The problem is the overlapping of access rules that grants you at the end the unlink. This has to be the way.

**Comment by pedrobaeza** [comment]
Merging

File: account_payment_return/models/payment_return.py (hunk 4 above)
Comment (yajo) [reply account_payment_return/models/payment_return.py:115]: Well, I know it's late and after all it works fine this way, but just to clarify, I meant a _global_ rule. Quoting from the above link:

> Global rules are subtractive, they must all be matched for a record to be accessible

Comment (pedrobaeza) [reply account_payment_return/models/payment_return.py:115]: OK, thanks for clarifying. Odoo doesn't do this kind of things by security rules, so there should be a reason for that. Maybe performance as pointed by Carlos, or maybe to not bypass this check removing access rule.

**Pull request merged**